src/prefect/tasks/kubernetes/secrets.py

```diff
@@ -0,0 +1,125 @@
+import base64
+
+from typing import Any, cast, Callable, Optional
+
+from kubernetes import client
+
+from prefect.tasks.secrets.base import SecretBase
+from prefect.utilities.tasks import defaults_from_attrs
+from prefect.utilities.kubernetes import get_kubernetes_client
+
+
+class KubernetesSecret(SecretBase):
+    """
+    Task for creating a Prefect secret from the kubernetes secret.
+
+    This task will read the secret object from kubernetes and extract the value of the secret_key
+    and decode it. The kubernetes secret can be created by sealedsecret or vault and read here.
+
+    Note that you need to ensure that you have the right RBAC from your cluster admin to read the secret
+    Note that all initialization arguments can optionally be provided or overwritten at runtime.
+
+    1. Attempt to use a Prefect Secret that contains a Kubernetes API Key. If
+    `kubernetes_api_key_secret` = `None` then it will attempt the next two connection
+    methods. By default the value is `KUBERNETES_API_KEY` so providing `None` acts as
+    an override for the remote connection.
+    2. Attempt in-cluster connection (will only work when running on a Pod in a cluster)
+    3. Attempt out-of-cluster connection using the default location for a kube config file
+
+
+    Args:
+        - secret_name (string, optional): The name of the kubernetes secret object
+        - secret_key (string, optional): The key to look for in the kubernetes data
+        - namespace (str, optional): The Kubernetes namespace to read the secret from
+        - kube_kwargs (dict, optional): Optional extra keyword arguments to pass to the
+            Kubernetes API (e.g. `{"pretty": "...", "dry_run": "..."}`)
+        - kubernetes_api_key_secret (str, optional): the name of the Prefect Secret
+            which stored your Kubernetes API Key; this Secret must be a string and in
+            BearerToken format
+        - cast_function (Callable[[Any], Any]): A function that will be called on the Parameter
+            value to coerce it to a type.
```

Comment:
```suggestion
        - cast_function (Callable[[Any], Any], optional): If provided, this will
            be called on the secret to transform it before returning. An example
            use might be passing in `json.loads` to load values from stored JSON.
```

Reply:
What do you think about renaming this to `cast` instead to maintain consistency with the api of `EnvVarSecret`?

Reply:
it was cast before, however I use ```cast``` from ```typing``` in this case it changes the function. This is why I had to give it another name

Reply:
I renamed the parameter to cast. My bad about the previous comment